CMakeLists.txt

```diff
@@ -156,6 +156,16 @@ ELSEIF(CMAKE_C_COMPILER_ID STREQUAL "Clang")
     SET(XNNPACK_ENABLE_AVX256VNNIGFNI OFF)
   ENDIF()
 ENDIF()
+OPTION(XNNPACK_ENABLE_AVX512VNNIGFNI "Build XNNPACK with AVX512VNNIGFNI micro-kernels" ON)
+IF(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+  IF(CMAKE_C_COMPILER_VERSION VERSION_LESS "11")
+    SET(XNNPACK_ENABLE_AVX512VNNIGFNI OFF)
+  ENDIF()
+ELSEIF(CMAKE_C_COMPILER_ID STREQUAL "Clang")
+  IF(CMAKE_C_COMPILER_VERSION VERSION_LESS "15")
```

Comment:
Have the versions of GCC and Clang that support GFNI and AVX512 been confirmed? I'm pretty sure these can be lower.

Reply:
No, this wasn't confirmed. Just using `AVX-VNNI` conservatively since this was disabled before.